vm/src/obj/objfloat.rs

```diff
@@ -667,4 +686,7 @@ impl TryFromObject for IntoPyFloat {
 #[rustfmt::skip] // to avoid line splitting
 pub fn init(context: &PyContext) {
     PyFloat::extend_class(context, &context.types.float_type);
+    extend_class!(context, &context.types.float_type, {
+        "__new__" => context.new_rustfunc(float_new),
+    });
```

Comment:
You should keep `float_new` in the `pyimpl`, it's a really simple function anyway.

Reply:
I do like the separation of `to_float` out into its own function though, nice work!

Reply:
I actually had no idea you could do that, neat!